src/olympia/devhub/tasks.py

```diff
@@ -187,9 +187,13 @@ def handle_upload_validation_result(results, upload_pk, channel):
     FileUpload instance."""
     upload = FileUpload.objects.get(pk=upload_pk)
 
-    if not upload.addon_id:
+    if (not upload.addon_id or
+            not upload.addon.find_latest_version(channel=channel, exclude=())):
```

Comment:
yeah, I just wasn't sure if it was a valid use case, or at least something developers commonly do, rather than just leaving the older versions as disabled.